networks/suzuka/suzuka-client/src/tests/settlement_state.rs

```diff
@@ -0,0 +1,290 @@
+use alloy_network::Ethereum;
+use alloy_network::EthereumSigner;
+use alloy_primitives::Address;
+use alloy_primitives::U256;
+use alloy_provider::Provider;
+use alloy_provider::ProviderBuilder;
+use alloy_signer_wallet::LocalWallet;
+use alloy_sol_types::sol;
+use alloy_transport::Transport;
+use aptos_sdk::{
+	coin_client::CoinClient,
+	rest_client::{Client, FaucetClient},
+	types::LocalAccount,
+};
+use std::env;
+use std::str::FromStr;
+use url::Url;
+
+sol!(
+	#[allow(missing_docs)]
+	#[sol(rpc)]
+	MCR,
+	"../../../protocol-units/settlement/mcr/client/abi/MCR.json"
+);
```

Comment:
This generates the RPC client from `MCR.json` (BTW, this relative path will break if the crate is ever vendored outside the workspace).
In `eth_client.rs`, the ABI is compiled from `MCRLegacy.json`. I wonder if we can run into a discrepancy?

Reply:
Both `eth_client.rs` and [settlement_state.rs](https://github.com/movementlabsxyz/movement/pull/105/files/71b53cf387f2357fb818e9ba7885486208b321c6#diff-415c7420ab8b4c2875c6cbe7cc79806fef15cb260512f70bac492e61ad4d9ec1) compile using the same file `MCR.json` . Perhaps I'm missing something but the intention was to use the same ABI file for all clients.

Reply:
Right, it got moved to `MCRLegacy.json` in main, and there's a bunch of new eth client code that will use the changed contract `MCR.json`, but it's not in use yet. I have updated the file path in my merge branch.